config/manifests/stable/local-storage-operator.clusterserviceversion.yaml

```diff
@@ -152,6 +152,9 @@ spec:
   minKubeVersion: 1.14.0
   provider:
     name: Red Hat
+  relatedImages:
+    - image: quay.io/openshift/origin-local-storage-mustgather:latest
+      name: local-storage-mustgather
```

Comment:
There is some magic here I don't understand, why do we need only the mustgather image listed under `relatedImages` out of the 4 images in the `image-references` file?
You mentioned the other images are populated by the ART pipeline, but why are those populated and not mustgather?

Reply:
There is some doc how `relatedImages` is populated: 
https://art-docs.engineering.redhat.com/olm-bundles/#image-references-and-specrelatedimages

Even after reading it I am not sure if they overwrite existing `relatedImages`.

Reply:
If I understand [this doc section](https://art-docs.engineering.redhat.com/olm-bundles/#image-references-and-specrelatedimages), you don't need to set `relatedImages` here because it will be populated by the contents of `image-references` anyway.

Reply:
It's not that clear to me

> Additionally, a [spec.relatedImages section](https://pkgs.devel.redhat.com/cgit/containers/vertical-pod-autoscaler-operator-bundle/tree/manifests/vertical-pod-autoscaler.clusterserviceversion.yaml?h=rhaos-4.12-rhel-8&id=0c2cc291be97889232d1c7e54ed5f576e4960cfd#n525) is created in the YAML listing all occurrences where CSV pullspecs were replaced by production pullspecs with digests

If there is no replacement in CSV, i.e. the image is not there at all, then it won't be added to `relatedImages`.

Reply:
Anyway, let's try it. I removed `relatedImages` from CSV, we'll see if ART populates it from related-images.

Reply:
ART did not populate it. From internal build:

```
  relatedImages:
  - image: registry.redhat.io/openshift4/ose-kube-rbac-proxy@sha256:ad39f40a0eb8865fa52ff2333b43d862cecb3443bfa3d43cfcfcea89468c7eaa
    name: ose-kube-rbac-proxy
  - image: registry.redhat.io/openshift4/ose-local-storage-diskmaker@sha256:8c849ae748a75dbca5022e7ae1f99930f112f427b70f293e5e8e659496c2c1f3
    name: ose-local-storage-diskmaker
  - image: registry.redhat.io/openshift4/ose-local-storage-operator@sha256:4b532669990f7acb02e1c51da935830e765a26e7cfe2ad0db5fc4d56d07553f8
    name: ose-local-storage-operator
```